easybuild/framework/easyblock.py

```diff
@@ -2584,9 +2584,24 @@ def check_checksums_for(self, ent, sub='', source_cnt=None):
         ec_fn = os.path.basename(self.cfg.path)
         checksum_issues = []
 
-        sources = ent.get('sources', [])
-        patches = ent.get('patches', []) + ent.get('postinstallpatches', [])
-        checksums = ent.get('checksums', [])
+        if isinstance(ent, EasyConfig):
+            # try to get value with templates resolved, but fall back to not resolving templates;
+            # this is better for error reporting that includes names of source files
+            try:
+                sources = ent.get('sources', [])
+                patches = ent.get('patches', []) + ent.get('postinstallpatches', [])
+                checksums = ent.get('checksums', [])
+            except EasyBuildError:
+                sources = ent.get_ref('sources')
+                patches = ent.get_ref('patches') + ent.get_ref('postinstallpatches')
+                checksums = ent.get_ref('checksums')
+        elif isinstance(ent, dict):
+            sources = ent.get('sources', [])
+            patches = ent.get('patches', []) + ent.get('postinstallpatches', [])
+            checksums = ent.get('checksums', [])
+        else:
+            raise EasyBuildError(f"Unexpected value type in EasyBlock.check_checksums_for: {type(ent)}")
+
```

Comment:
Can we simplify to have a consistent behaviour across the board? Is there any downside to always fallback to not resolved templates?
```suggestion
        # try to get value with templates resolved, but fall back to not resolving templates;
        # this is better for error reporting that includes names of source files
        try:
            sources = ent.get('sources', [])
            patches = ent.get('patches', []) + ent.get('postinstallpatches', [])
            checksums = ent.get('checksums', [])
        except EasyBuildError:
            sources = ent.get_ref('sources')
            patches = ent.get_ref('patches') + ent.get_ref('postinstallpatches')
            checksums = ent.get_ref('checksums')

```

Reply:
There's a big downside here: `ent` may be a `dict` value rather than an `EasyConfig` instance, and a `dict` value doesn't support `.get_ref`, so that would lead to a hard crash...

That's part of the reason why I enhanced the test for `check_checksums_for`

Reply:
Ok, then I suggest to rewrite the same logic as follows to simplify this code:
```suggestion
        # try to get value with templates resolved, but fall back to not resolving templates;
        # this is better for error reporting that includes names of source files
        try:
            sources = ent.get('sources', [])
            patches = ent.get('patches', []) + ent.get('postinstallpatches', [])
            checksums = ent.get('checksums', [])
        except EasyBuildError as err:
            if isinstance(ent, EasyConfig):
                sources = ent.get_ref('sources')
                patches = ent.get_ref('patches') + ent.get_ref('postinstallpatches')
                checksums = ent.get_ref('checksums')

```

Reply:
fixed in 85987c298